lib/ja_serializer/builder/included.ex

```diff
@@ -77,14 +78,28 @@ defmodule JaSerializer.Builder.Included do
       context
       |> get_data(definition)
       |> List.wrap()
-      |> resource_objects_for(context.conn, definition.serializer, child_opts)
       |> Enum.reduce({[], included}, fn item, {cont, included} ->
-        key = resource_key(item)
-
-        if MapSet.member?(known, key) or Map.has_key?(included, key) do
-          {cont, included}
+        if is_map(item) do
+          key =
+            {definition.serializer.id(item, context.conn),
+             definition.serializer.type(item, context.conn)}
```

Comment:
What's the rationale behind doing this rather than using `resource_key/1`? It looks like there's one other spot in this file that calls `resource_key/1`, should that be updated as well? Sorry, naive questions, but I don't know this codebase at all.

Reply:
`resource_key` builds the `{id, type}` pair based on a `JaSerializer.Builder.ResourceObject`, which is what we're trying to avoid building. Instead, we do what would have happened, which is check with the serializer to get a type and ID based on the item.

Reply:
Ah, thanks, that makes sense.